core/controllers/base.py

```diff
@@ -707,30 +747,48 @@ def handle_exception(self, exception, unused_debug_mode):
         if isinstance(exception, self.PageNotFoundException):
             logging.warning('Invalid URL requested: %s', self.request.uri)
             self.error(404)
-            self._render_exception(
-                404, {
-                    'error': 'Could not find the page %s.' % self.request.uri})
+            values = {
+                'error': 'Could not find the page %s.' % self.request.uri,
+                'status_code': 404
+            }
+            self._render_exception(404, values)
             return
 
         logging.exception('Exception raised: %s', exception)
 
         if isinstance(exception, self.UnauthorizedUserException):
             self.error(401)
-            self._render_exception(401, {'error': str(exception)})
+            values = {
+                'error': str(exception),
+                'status_code': 401
+            }
+            self._render_exception(401, values)
             return
 
         if isinstance(exception, self.InvalidInputException):
             self.error(400)
-            self._render_exception(400, {'error': str(exception)})
+            values = {
+                'error': str(exception),
+                'status_code': 400
+            }
+            self._render_exception(400, values)
             return
 
         if isinstance(exception, self.InternalErrorException):
             self.error(500)
-            self._render_exception(500, {'error': str(exception)})
+            values = {
+                'error': str(exception),
+                'status_code': 500
+            }
+            self._render_exception(500, values)
             return
 
         self.error(500)
-        self._render_exception(500, {'error': str(exception)})
+        values = {
+            'error': str(exception),
+            'status_code': 500
```

Comment:
Have we incorrectly been omitting the status_code all this while? (Where did you see that it was needed?)

Slightly confused since AFAICT we pass it as the first argument to _render_exception().

Reply:
https://github.com/oppia/oppia/blob/900a25a62a37057c4779247cf9c9c0f14b66e788/core/controllers/base.py#L710

Till now (before this PR's changes), we are first defining a Dict[str, str] value and passing it to `_render_exception` but in `_render_exception` function after asserting the error code we are adding an int value to the same Dict ( Dict having Dict[str, str] value ) so mismatch in typing occurs. 
https://github.com/oppia/oppia/blob/a09b592e76daf7af1818928d89a302085609f99c/core/controllers/base.py#L649
`First, we are doing Dct[str, st] and then we are adding int value. `

So what I try to do here is -- defining a dict in the first place with proper TypedDict structure ( before passing to `_render_exception` function as an argument) rather than passing the error code and Dict[str, str] individually. And instead of asserting the error code explicitly, I'm asserting the dict's error code value.

Reply:
I hope that clears the thing up, but please let me know If you need more info, I'll try again to explain, Thanks!